Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+---
+title: BestPractices
+weight: 3950
+indent: true
+---
+
+# Rook-Ceph Best Practices
+
+## Overview
+
+Ceph and Kubernetes are both complex tools and harmonizing the interactions between the two can be
+daunting. This is especially true for users who are new to operating either system, prompting
+questions such as:
+ * How can I restrict Ceph to a portion of my nodes?
+ * Can I set Kubernetes CPU or RAM limits for my Ceph daemons?
+ * What are some ways to get better performance from my cluster?
+
+This document covers tested patterns and best practices to answer these questions and more. Our
+examples will help you configure and manage your Ceph cluster running in Kubernetes to meet your
+needs. The following examples and advice are based on Ceph Octopus (v15) with Rook v1.3 running in a
+Kubernetes 1.17 cluster.
+
+This is a moderately advanced topic, so basic experience with Rook is recommended. Before you begin,
+ensure you have the following requisite knowledge:
+ * Basics of Kubernetes
+ * How to create Kubernetes applications using manifests
+ * Kubernetes topics:
+   - Pods
+   - Nodes
+   - Labels
+   - Topology
+   - Taints and tolerations
+   - Affinity and Anti-affinity
+   - Resource requests
+   - Limits
+ * Ceph components and daemons, basic Ceph configuration
+ * Rook basics and how to install Rook-Ceph
+
+In places, we will give examples that describe an imaginary data center. This data center is
+hypothetical, and it will focus on the Ceph- and Rook-centric elements and ignore user applications.
+
+Our example data center has two rooms for data storage. A properly fault tolerant Ceph cluster
```

Comment:
My goal was to introduce a single example we could use to demonstrate best practices that wasn't the most trivial example. I have always found it extremely irritating when software is documented assuming the simplest case because that doesn't really help understand how to change things for more advanced situations. I'd rather keep the advanced situation here, though I can clarify this intent in the doc.

Reply:
Yep, keeping this as an advanced walkthrough sounds good with the changed title.